src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/TargetFrameworkMoniker.cs

```diff
@@ -2,25 +2,200 @@
+            init => _framework = value;
+        }
 
-        public bool IsWindows => Name.Contains("windows");
+        public Version FrameworkVersion
+        {
+            get => NormalizeVersion(_frameworkVersion) ?? _frameworkVersion;
+            init => _frameworkVersion = value;
+        }
+
+        public string? Platform
+        {
+            get
+            {
+                if (IsWindows)
+                {
+                    return Platforms.Windows;
+                }
+
+                if (string.IsNullOrEmpty(_platform))
+                {
+                    return null;
+                }
+
+                return _platform;
+            }
+
+            init => _platform = value;
+        }
+
+        public Version? PlatformVersion
+        {
+            get => NormalizeVersion(_platformVersion);
+            init => _platformVersion = value;
+        }
+
+        public string Name => ToString();
+
+        public virtual bool Equals(TargetFrameworkMoniker? other)
+        {
+            if (other is null)
+            {
+                return false;
+            }
+
+            return string.Equals(Framework, other.Framework, StringComparison.OrdinalIgnoreCase)
+                && Equals(FrameworkVersion, other.FrameworkVersion)
+                && string.Equals(Platform, other.Platform, StringComparison.OrdinalIgnoreCase)
+                && Equals(PlatformVersion, other.PlatformVersion);
+        }
+
+        public override int GetHashCode()
+        {
+            var hashcode = default(HashCode);
+
+            hashcode.Add(Framework, StringComparer.OrdinalIgnoreCase);
+            hashcode.Add(FrameworkVersion);
+            hashcode.Add(Platform, StringComparer.OrdinalIgnoreCase);
+            hashcode.Add(PlatformVersion);
+
+            return hashcode.ToHashCode();
+        }
+
+        public override string ToString()
+        {
+            var sb = new StringBuilder();
+
+            sb.Append(Framework);
+            sb.Append(FrameworkVersion);
+
+            if (IsFramework)
+            {
+                sb.Replace(".", string.Empty);
+            }
+            else if (IsNet50OrAbove && Platform is string platform)
+            {
+                sb.Append('-');
+                sb.Append(platform);
+
+                if (PlatformVersion is Version platformVersion)
+                {
+                    sb.Append(platformVersion);
+                }
+            }
+
+            return sb.ToString();
+        }
+
+        public bool IsFramework
+        {
+            get
+            {
+                if (_framework.Equals(Net, StringComparison.OrdinalIgnoreCase) && !Is50OrAbove)
+                {
+                    return true;
+                }
+
+                return _framework.ToUpperInvariant().Contains(NetFramework.ToUpperInvariant());
+            }
+        }
+
+        public bool IsNetStandard => _framework.ToUpperInvariant().Contains(NetStandard.ToUpperInvariant());
```

Comment:
nit: should be `.Equals` now that framework is a distinct field

Reply:
This is to potentially match `.NETStandard`, as well as `netstandard` - both show up in NuGet parsing

Reply:
I've added a comment to that effect.